woocommerce/woocommerce
# Added coupon to API package and related tests

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds coupons to the `api` package. It also includes an e2e test to verify that the API behaves as expected. Additionally, this renames some of the tests and adds a new API testing grouping to `tests/e2e/core-tests/specs/index.js`.

Closes https://github.com/woocommerce/woocommerce/issues/28882 .

### How to test the changes in this Pull Request:

1. Verify the CI tests run and pass
2. Run the tests locally and note the API tests:
<img width="502" alt="Screen Shot 2021-03-01 at 11 39 20 AM" src="https://user-images.githubusercontent.com/71906536/109542802-c8e94500-7a82-11eb-8058-735d8b522928.png">

3. Comment out the delete step in the test and edit the `coupon` in `tests/e2e/config/default.json` to update or add any values that are available in the [Coupons API](https://woocommerce.github.io/woocommerce-rest-api-docs/#coupons). Also make sure to update `Settings > Permalinks` to be "Post name".
4. Manually run the test `npx wc-e2e test:e2e tests/e2e/specs/rest-api/coupon.js`
5. Repeat steps 3 and 4 with different combinations. Check under `Marketing > Coupons` and verify the coupon was created correctly. Note: you will need to update the coupon code each time, or delete each coupon manually after it's been created otherwise a "coupon already exists" error will occur.

### Other information:


### Changelog entry

N/A
